internal/backend/backend.go

```diff
@@ -263,3 +270,23 @@ type upsertFailure struct {
 	code    int
 	added   *int64
 }
+
+func (m *metrics) Collect(ch chan<- prometheus.Metric) {
+	age := math.Inf(0)
+	if m.oldest != nil {
+		age = float64(now().Unix() - *m.oldest)
+	}
+
+	ch <- prometheus.MustNewConstMetric(m.oldestFailureAge, prometheus.GaugeValue, age)
+
+	m.oldestFailureTimestamp.Collect(ch)
+	m.retries.Collect(ch)
```

Comment:
I think I'd prefer to define a type just for the kubernetes_scanner_backend_oldest_failure_age metric and implement the Collector just for that, rather than couple the registration of the more "normal" metrics to the fact that we've felt the need to implement a custom collector for one of them.

Reply:
We paired on this, given that there is coupling between the oldest failure timestamp and age metrics, a custom collector that couples those 2 makes sense. At that point it is less odd that the other unrelated metrics are part of it. Those other 2 could be unbundled, but at this point I'm downgrading this thread to a nit 😅